packages/http-server/src/server.ts

```diff
@@ -155,28 +155,38 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
     )();
   };
 
+  function setCommonCORSHeaders(incomingHeaders: IncomingHttpHeaders, res: ServerResponse) {
+    res.setHeader('Access-Control-Allow-Origin', incomingHeaders['origin'] || '*');
+    res.setHeader('Access-Control-Allow-Headers', incomingHeaders['access-control-request-headers'] || '*');
+    res.setHeader('Access-Control-Allow-Credentials', 'true');
+    res.setHeader('Access-Control-Expose-Headers', incomingHeaders['access-control-expose-headers'] || '*');
+  }
+
   const server = micri(
     Router.router(
       Router.on.options(
         () => opts.cors,
         (req: IncomingMessage, res: ServerResponse) => {
-          res.setHeader('Access-Control-Allow-Origin', req.headers['origin'] || '*');
-          res.setHeader('Access-Control-Allow-Headers', req.headers['access-control-request-headers'] || '*');
-          res.setHeader('Access-Control-Allow-Credentials', 'true');
-          res.setHeader('Access-Control-Allow-Methods', 'GET,DELETE,HEAD,PATCH,POST,PUT');
-          res.setHeader('Access-Control-Expose-Headers', req.headers['access-control-expose-headers'] || '*');
-          res.setHeader('Vary', 'origin');
-          res.setHeader('Content-Length', '0');
-          send(res, 204);
+          setCommonCORSHeaders(req.headers, res);
+
+          if (!!req.headers['origin'] && !!req.headers['access-control-request-method']) {
+            // This is a preflight request, so we'll respond with the appropriate CORS response
+            res.setHeader(
+              'Access-Control-Allow-Methods',
+              req.headers['access-control-request-method'] || 'GET,DELETE,HEAD,PATCH,POST,PUT,OPTIONS'
+            );
+
+            res.setHeader('Vary', 'origin');
+            res.setHeader('Content-Length', '0');
```

Comment:
This is not required:
>    1.  Any response to a HEAD request and any response with a 1xx
>       (Informational), 204 (No Content), or 304 (Not Modified) status
>       code is always terminated by the first empty line after the
>       header fields, regardless of the header fields present in the
>       message, and thus cannot contain a message body.

Source: https://tools.ietf.org/html/rfc7230#section-3.3.3

Reply:
This is required because of Safari. For some reason, even though we're sending a 204, Safari will check the content length anyway and have trouble if there's not.

https://github.com/pallets/werkzeug/issues/1025#issuecomment-390469386

Reply:
Got it, how about commenting this line?

Reply:
Got it, how about commenting this line?

Reply:
> 
> 
> This is required because of Safari. For some reason, even though we're sending a 204, Safari will check the content length anyway and have trouble if there's not.
> 
> [pallets/werkzeug#1025 (comment)](https://github.com/pallets/werkzeug/issues/1025#issuecomment-390469386)

Doesn't the quoted comment say the exact opposite?
> the **unwanted Content-Length header** currently makes Mac OS Safari throw a network error and **fail, when a 204 is returned**
 (emphasis mine)

Reply:
I am sorry I pointed you to the wrong resource.

https://github.com/fastify/fastify-cors/blob/master/index.js#L88-L89

And yes I confirm such behaviour in Safari 13.0.5